api/src/main/java/org/openbaton/nfvo/api/catalogue/RestVNFPackage.java

```diff
@@ -87,7 +93,136 @@ public String onboard(
 
     log.debug("Onboarding");
     if (!file.isEmpty()) {
-      byte[] bytes = file.getBytes();
+
+      byte[] bytes = new byte[0];
+      Random random = new Random();
+      int ran = random.nextInt();
+      String directory = "/tmp/random" + ran + "/";
+
+      // write the file to the filesystem
+      File tempDirectory = new File(directory);
+      if (!tempDirectory.exists()) {
+        tempDirectory.mkdir();
+      }
+
+      String name = file.getOriginalFilename();
+      File tarFile = new File(directory + name);
+      file.transferTo(tarFile);
+      log.debug("File size is " + file.getSize());
+
+      Runtime rt = Runtime.getRuntime();
+
+      try {
+        // untar the file
+        Process p1 =
+            rt.exec(new String[] {"bash", "-c", "tar -xf " + name}, new String[0], tempDirectory);
```

Comment:
@RuthDVerizon I'm a bit worried about using bash commands directly in the NFVO. Main reason is that there are some users using windows, and this particular command will probably fail and throw some exceptions. 
In addition, I see that some parts of this code is doing more or less the same which is done in the VNFPackageManagement ( https://github.com/openbaton/NFVO/blob/8e7206efa6a72ad315725062d4bb9019e64e9afe/core/core-impl/src/main/java/org/openbaton/nfvo/core/api/VNFPackageManagement.java#L105) class. In that case, we are making use of some java libraries for extracting content from the VNF Package, without requiring the execution of bash commands.